pinot-clients/pinot-java-client/src/main/java/org/apache/pinot/client/Connection.java

```diff
@@ -190,8 +192,10 @@ public Future<ResultSetGroup> executeAsync(@Nullable String tableName, String qu
   private static String[] resolveTableName(String query) {
     try {
       SqlNodeAndOptions sqlNodeAndOptions = CalciteSqlParser.compileToSqlNodeAndOptions(query);
-      List<String> tableNames = CalciteSqlParser.extractTableNamesFromNode(sqlNodeAndOptions.getSqlNode());
-      return tableNames.toArray(new String[0]);
+      String tableName = (sqlNodeAndOptions.getSqlNode() != null
+          ? RequestUtils.getTableName(CalciteSqlParser.compileSqlNodeToPinotQuery(sqlNodeAndOptions.getSqlNode()))
+          : CalciteSqlCompiler.compileToBrokerRequest(query).getQuerySource().getTableName());
+      return new String[]{tableName};
     } catch (Exception e) {
       LOGGER.error("Cannot parse table name from query: {}", query, e);
```

Comment:
nit we can add additional comments
```suggestion
      LOGGER.error("Cannot parse table name from query: {}. Fallback to broker selector default.", query, e);
```